packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -206,77 +196,43 @@ async function instantiateModule(
   return Object.freeze(ssrModule)
 }
 
+const isDedupable = (id: string) =>
+  bareImportRE.test(id) && !(isBuiltin(id) || id.endsWith('.node'))
+
 // In node@12+ we can use dynamic import to load CJS and ESM
 async function nodeImport(
   id: string,
   importer: string,
-  resolveOptions: InternalResolveOptions
+  dedupe?: string[],
+  projectRoot?: string
 ) {
-  // Node's module resolution is hi-jacked so Vite can ensure the
-  // configured `resolve.dedupe` and `mode` options are respected.
-  const viteResolve = (
-    id: string,
-    importer: string,
-    options = resolveOptions
-  ) => {
-    const resolved = tryNodeResolve(id, importer, options, false)
-    if (!resolved) {
-      const err: any = new Error(
-        `Cannot find module '${id}' imported from '${importer}'`
-      )
-      err.code = 'ERR_MODULE_NOT_FOUND'
-      throw err
-    }
-    return resolved.id
-  }
+  let unhookNodeResolve: (() => void) | undefined
+  if (dedupe && projectRoot) {
+    const resolve = Module.createRequire(projectRoot + '/index.js')
+    const isLinkedModule = (module: NodeModule) =>
+      !module.id.startsWith(projectRoot + '/')
```

Comment:
Unsure if this check if compatible with Windows..

Reply:
good question. I'd love to have 2 things (though that's better discussed on discord)
1) hopefully get rid of windows path issues once and for all by using `pathe` by @pi0 
2) a complete util for conversion of id to path to url, usable within vite and plugins alike (that also covers working with url params, eg by exposing url as URL, or maybe typing params as a map with a few blessed keys (t, raw etc).

Reply:
we should discuss this with Patak when he's back from vacation 👍